hudi-spark-datasource/hudi-spark3.3.x/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/Spark332PlusHoodieParquetFileFormat.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.parquet
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.datasources.PartitionedFile
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.sources.Filter
+import org.apache.spark.sql.types.StructType
+
+class Spark332PlusHoodieParquetFileFormat(override protected val shouldAppendPartitionValues: Boolean) extends Spark32PlusHoodieParquetFileFormat(shouldAppendPartitionValues) {
```

Comment:
With this class under `hudi-spark3.3.x`, Hudi won't be able to compile with Spark 3.3.1 anymore

Reply:
Yes this is true since, but my understanding is currently that hudi supports only minor version of spark. https://github.com/apache/hudi/blob/master/pom.xml#L159 im not sure if users should be changing the version in the poms and trying to compile with a different version. cc @yihua 

Currently when im taking the open source jars from Hudi which are compiled via spark 3.3.1 and try running them on spark 3.3.2 they hit failures due to the new config causing issues, so we eventually I think have to support this. 

However I think the main thing we need to check is runtime support meaning if this change of upgrading to spark 3.3.2 atleast can run on a Spark 3.3.1 runtime so we dont break users.

Reply:
Update on this when compiling with spark 3.3.1 things are working fine even with these 3.3.2 changes .
`mvn clean install  -Dspark3 -Dscala-2.12 -DskipTests  -pl packaging/hudi-spark-bundle -am -Dspark.version=3.3.1` 

```
[INFO] Hudi ............................................... SUCCESS [  2.629 s]
[INFO] hudi-tests-common .................................. SUCCESS [  1.752 s]
[INFO] hudi-common ........................................ SUCCESS [ 29.852 s]
[INFO] hudi-hadoop-mr ..................................... SUCCESS [  7.301 s]
[INFO] hudi-sync-common ................................... SUCCESS [  2.429 s]
[INFO] hudi-hive-sync ..................................... SUCCESS [  6.965 s]
[INFO] hudi-aws ........................................... SUCCESS [  4.255 s]
[INFO] hudi-timeline-service .............................. SUCCESS [  2.550 s]
[INFO] hudi-client ........................................ SUCCESS [  0.082 s]
[INFO] hudi-client-common ................................. SUCCESS [ 13.594 s]
[INFO] hudi-spark-client .................................. SUCCESS [ 36.826 s]
[INFO] hudi-spark-datasource .............................. SUCCESS [  0.079 s]
[INFO] hudi-spark-common_2.12 ............................. SUCCESS [ 28.910 s]
[INFO] hudi-spark3-common ................................. SUCCESS [  9.304 s]
[INFO] hudi-spark3.2plus-common ........................... SUCCESS [ 12.118 s]
[INFO] hudi-spark3.3.x_2.12 ............................... SUCCESS [ 23.851 s]
[INFO] hudi-java-client ................................... SUCCESS [  4.792 s]
[INFO] hudi-spark_2.12 .................................... SUCCESS [ 53.643 s]
[INFO] hudi-spark3.3-bundle_2.12 .......................... SUCCESS [ 29.585 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  04:31 min
[INFO] Finished at: 2023-04-19T11:56:41-07:00
[INFO] ---------------------------------------------

```
confirmed via mvn logs that it was using 3.3.1
```
[INFO] Copying derby-10.14.2.0.jar to /Users/rchertar/hudi/hudi-spark-datasource/hudi-spark/target/lib/derby-10.14.2.0.jar
[INFO] Copying spark-sql_2.12-3.3.1-tests.jar to /Users/rchertar/hudi/hudi-spark-datasource/hudi-spark/target/lib/spark-sql_2.12-3.3.1-tests.jar
[INFO] Copying spark-core_2.12-3.3.1-tests.jar to /Users/rchertar/hudi/hudi-spark-datasource/hudi-spark/target/lib/spark-core_2.12-3.3.1-tests.jar
[INFO] Copying spark-catalyst_2.12-3.3.1-tests.jar to /Users/rchertar/hudi/hudi-spark-datasource/hudi-spark/target/lib/spark-catalyst_2.12-3.3.1-tests.jar
[INFO] Copying scala-parser-combinators_2.12-1.1.2.jar to /Users/rchertar/hudi/hudi-spark-datasource/hudi-spark/target/lib/scala-parser-combinators_2.12-1.1.2.jar
```

Reply:
I'm good if we are not running into any compilation issues

Reply:
The log you show is different; it copies the jars to `target/lib`.  Could you try modifying the Spark version directly in POM and see if the changes can be compiled with Spark 3.3.1, to be safe?